paasta_tools/tron_tools.py

```diff
@@ -651,10 +685,24 @@ def format_tron_action_dict(action_config: TronActionConfig, use_k8s: bool = Fal
             and action_config.get_iam_role()
             and not action_config.for_validation
         ):
+            # this service account will be used for normal Tron batches as well as for Spark executors
             result["service_account_name"] = create_or_find_service_account_name(
-                iam_role=action_config.get_iam_role(), namespace="tron"
+                iam_role=action_config.get_iam_role(),
+                namespace=EXECUTOR_TYPE_TO_NAMESPACE[executor],
```

Comment:
Right now EXECUTOR_TYPE_TO_NAMESPACE maps `paasta` executors->`paasta` namespace, but here we see we used to create them in `tron` namespace, not `paasta`. 

Shoudl the `EXECUTOR_TYPE_TO_NAMESPACE` map get updated to return the tron namesapce instead?

Reply:
doh, yup - i wasn't thinking too good here. lemme also add a test for this if its not too painful

Reply:
actually, lemme write a test for this once i refactor to send the namespace to tron since that'll make this easier :p